lib/sles4sap/azure_cli.pm

```diff
@@ -1374,3 +1378,163 @@ sub az_resource_delete {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    $args{timeout} //= '90';
+
+    my $az_cmd = join(' ',
+        'az storage blob upload',
+        '--only-show-errors',
+        "--container-name $args{container_name}",
+        "--account-name $args{storage_account_name}",
+        "--file $args{file}"
+    );
+    assert_script_run(join(' ', $az_cmd), timeout => $args{timeout});
+}
+
+=head2 az_storage_blob_lease_acquire
+
+    az_storage_blob_lease_acquire(
+        container_name=>'somecontainer',
+        storage_account_name=>'storageaccount',
+        blob_name => 'somefilename' [, lease_duration=>'42']
+    );
+
+Acquire a lease for a storage blob.
+
+=over 4
+
+=item B<container_name> Existing storage container name.
+
+=item B<storage_account_name> Storage account name.
+
+=item B<blob_name> Blob name to acquire lease for.
+
+=item B<lease_duration> Lease duration between 15-60s. Default: infinite
+
+=back
+=cut
+
+sub az_storage_blob_lease_acquire {
+    my (%args) = @_;
+    foreach ('container_name', 'storage_account_name', 'blob_name') {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    $args{lease_duration} //= '-1';    # -1 = infinite lease
+
+    my $az_cmd = join(' ',
+        'az storage blob lease acquire',
+        '--only-show-errors',
+        "--container-name $args{container_name}",
+        "--account-name $args{storage_account_name}",
+        "--blob-name $args{blob_name}",
+        "--lease-duration $args{lease_duration}",
+        '--output tsv'    # Json output won't work here.
+    );
+
+    my $lease_id = script_output($az_cmd, proceed_on_failure => 1);
+    return ($lease_id) if qesap_az_validate_uuid_pattern($lease_id);    # Return only valid output.
+}
+
+=head2 az_storage_blob_list
+
+    az_storage_blob_list(
+        container_name=>'somecontainer',
+        storage_account_name=>'storageaccount' [, query=>'[].name']
+    );
+
+List information about storage blob(s) specified by B<storage_account_name>, B<container_name> and B<query>.
+
+=over 3
+
+=item B<container_name> Existing storage container name.
+
+=item B<storage_account_name> Storage account name.
+
+=item B<query> Query in jmespath format
+
+=back
+=cut
+
+sub az_storage_blob_list {
+    my (%args) = @_;
+    foreach ('container_name', 'storage_account_name') {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    $args{query} //= '[].name';
+
+    my $az_cmd = join(' ',
+        'az storage blob list',
+        '--only-show-errors',
+        "--container-name $args{container_name}",
+        "--account-name $args{storage_account_name}",
+        "--query \"$args{query}\"",
+        '--output json'
+    );
+
+    return decode_json(script_output($az_cmd));
+}
+
+=head2 az_storage_blob_update
+
+    az_storage_blob_update(container_name=>'container', account_name=>'stuff', name='blobby' [, lease_id=42]);
+
+Update properties of storage blob.
```

Comment:
say something about the return. Seems you are intentionally using `script_run` instead of `assert_script_run` to return the exit code and let the caller to decide. It is fine but makes sense to document it.

Reply:
Actually... that is a good point... I wanted to use `assert_script_run` just to make it fail, but I'll leave it as it is and make the caller decide.  
I need to fix the other part of the code that calls this function.

Reply:
I just updated the desc.